lib/elements/dom-repeat.js

```diff
@@ -542,18 +549,16 @@ export class DomRepeat extends domRepeatBase {
     const isntIdxToItemsIdx = this.__sortAndFilterItems(items);
     // If we're chunking, increase the limit if there are new instances to
     // create and schedule the next chunk
-    if (this.initialCount) {
-      this.__updateLimit(isntIdxToItemsIdx.length);
-    }
+    this.__updateLimit(isntIdxToItemsIdx.length);
     // Create, update, and/or remove instances
     this.__updateInstances(items, isntIdxToItemsIdx);
-    // Set rendered item count
-    this._setRenderedItemCount(this.__instances.length);
     // If we're chunking, schedule a rAF task to measure/continue chunking
     if (this.initialCount &&
        (this.__shouldMeasureChunk || this.__shouldContinueChunking)) {
       this.__debounceRender(this.__continueChunkingAfterRaf);
     }
+    // Set rendered item count
+    this._setRenderedItemCount(this.__instances.length);
```

Comment:
Since it's a notifying property, just noticed that it could have the same risk as debouncing the chunking after sending the do-change notification. Basically, do all the notifications last.